lib/mobility-core/src/Kernel/Storage/Clickhouse/Queries.hs

```diff
@@ -42,37 +50,39 @@ __orderBy = addClause
 __offset :: String -> Offset -> String
 __offset = addClause
 
-runClickhouse :: (MonadFlow m, ClickhouseFlow m env, FromJSON a) => (HttpConnection -> IO (Either String a)) -> m (Either String a)
-runClickhouse action = do
-  con <- asks (.clickhouseEnv.connection)
+runClickhouse :: (MonadFlow m, ClickhouseFlow m env, FromJSON a) => (HttpConnection -> IO (Either String a)) -> ClickhouseDb -> m (Either String a)
+runClickhouse action db = do
+  con <- case db of
+    ATLAS_DRIVER_OFFER_BPP -> asks (.driverClickhouseEnv.connection)
+    ATLAS_KAFKA -> asks (.kafkaClickhouseEnv.connection)
   L.runIO $ action con
 
-runRawQuery :: (MonadFlow m, ClickhouseFlow m env, FromJSON a) => String -> m (Either String a)
+runRawQuery :: (MonadFlow m, ClickhouseFlow m env, FromJSON a) => String -> ClickhouseDb -> m (Either String a)
 runRawQuery query = runClickhouse (`runQuery` getJSON query)
 
-runRawQuery' :: (MonadFlow m, ClickhouseFlow m env, FromJSON a) => ClickhouseExpr -> m (Either String a)
-runRawQuery' (ExprStr query) = runRawQuery query
-runRawQuery' _ = throwError $ InternalError "can't call this function with unresolved clickhouseExpr"
+runRawQuery' :: (MonadFlow m, ClickhouseFlow m env, FromJSON a) => ClickhouseExpr -> ClickhouseDb -> m (Either String a)
+runRawQuery' (ExprStr query) db = runRawQuery query db
+runRawQuery' _ _ = throwError $ InternalError "can't call this function with unresolved clickhouseExpr"
 
-constructQuery :: Typeable a => Proxy a -> ClickhouseExpr -> Maybe Limit -> Maybe Offset -> Maybe Order -> String
-constructQuery proxyTable expr mbLimit mbOffset mbOrder = do
-  addOffset . addLimit . addOrder $ __select proxyTable expr
+constructQuery :: Typeable a => Proxy a -> [Text] -> Maybe String -> ClickhouseExpr -> Maybe Limit -> Maybe Offset -> Maybe Order -> String
+constructQuery proxyTable columns maybeGroupBy expr mbLimit mbOffset mbOrder = do
+  addOffset . addLimit . addOrder $ __select proxyTable columns maybeGroupBy expr
   where
     addLimit q = maybe q (__limit q) mbLimit
     addOrder q = maybe q (__orderBy q) mbOrder
     addOffset q = maybe q (__offset q) mbOffset
 
-findAll :: (MonadFlow m, ClickhouseFlow m env, Typeable a, FromJSON a) => Proxy a -> ClickhouseExpr -> Maybe Limit -> Maybe Offset -> Maybe Order -> m (Either String [a])
-findAll proxyTable expr mbLimit mbOffset mbOrder = runRawQuery $ constructQuery proxyTable expr mbLimit mbOffset mbOrder
+findAll :: (MonadFlow m, ClickhouseFlow m env, Typeable a, FromJSON a) => ClickhouseDb -> Proxy a -> [Text] -> Maybe String -> ClickhouseExpr -> Maybe Limit -> Maybe Offset -> Maybe Order -> m (Either String [a])
+findAll db proxyTable columns maybeGroupBy expr mbLimit mbOffset mbOrder = runRawQuery (constructQuery proxyTable columns maybeGroupBy expr mbLimit mbOffset mbOrder) db
```

Comment:
Can we avoid passing db in everyQuery ? You can create a custom type class and create its instance for the return type you expect for a query, and it can stores information of which db it belong to.

And later let's plan a cleaner way to construct the query also.